### ISSUE TITLE
Schema mismatch in chats 

### ISSUE THREAD

**Issue by koji-m** [singer-io/tap-zendesk-chat#38]
While running extraction from `chats` in Stitch, following error (schema mismatch) occurs.

```
2021-12-16 01:04:31,126Z tap - CRITICAL Errors during transform

2021-12-16 01:04:31,126Z tap - webpath.0: data does not match {'properties': {'from': {'type': ['string', 'null']}, 'to': {'type': ['string', 'null']}, 'title': {'type': ['string', 'null']}, 'timestamp': {'format': 'date-time', 'type': ['string', 'null']}}, 'type': ['object', 'null']}

...

2021-12-16 01:04:31,127Z tap - singer.transform.SchemaMismatch: Errors during transform
```
I can see this error has been occuring continuously from Dec 9. (I'm running the extraction job hourly.)

**Comment by BenoitHugonnard** [comment]
If I'm not mistaken, it seems that `webpath` property is expected to be an array and it can sometimes be just a dict ?
```python
{
  'properties': {
    'from': {
      'type': [
        'string',
        'null'
      ]
    },
    'title': {
      'type': [
        'string',
        'null'
      ]
    },
    'to': {
      'type': [
        'string',
        'null'
      ]
    },
    'timestamp': {
      'format': 'date-time',
      'type': [
        'string',
        'null'
      ]
    }
  },
  'type': [
    'object',
    'null'
  ]
},
webpath: datadoesnotmatch{
  'items': {
    'properties': {
      'from': {
        'type': [
          'string',
          'null'
        ]
      },
      'title': {
        'type': [
          'string',
          'null'
        ]
      },
      'to': {
        'type': [
          'string',
          'null'
        ]
      },
      'timestamp': {
        'format': 'date-time',
        'type': [
          'string',
          'null'
        ]
      }
    },
    'type': [
      'object',
      'null'
    ]
  },
  'selected': True,
  'inclusion': 'available',
  'type': [
    'array',
    'null'
  ]
}
```

**Comment by syed-hamza-alee** [comment]
Facing same issue while running extraction from `chats` in Stitch

**Comment by bharathiatnoths** [comment]
Did anyone resolve this issue? I have same problem with zendesk chat integration

**Comment by haiderwain1** [comment]
Facing same issue when extracting from chats. Any update?

**Comment by vorpal56** [comment]
The fix is to change the webpath property in the catalog that's generated in discovery mode. This is likely due to some changes on Zendesk's part, though not entirely sure.

```
"webpath": {
    "type": ["null", "array"],
    "items": {
        "type": ["null", "object", "array"],
        "properties": {
            "timestamp": {
                "type": ["null", "string"],
                "format": "date-time"
            },
            "to": {
                "type": ["null", "string"]
            },
            "from": {
                "type": ["null", "string"]
            },
            "title": {
                "type": ["null", "string"]
            }
        },
        "items": {
            "type": ["null", "string", "integer"]
        },
        "additionalProperties": false
    }
}
```

**Comment by Vi6hal** [comment]
Hi @koji-m 
this issue is fixed in the following P.R https://github.com/singer-io/tap-zendesk-chat/pull/41
and released in version 0.3.2
